ambariclient/models.py

```diff
@@ -738,13 +760,109 @@ def restart(self, component_names=None):
             }))
         return self
 
+    def start(self, component_names=None):
+        """Starts this service."""
+        self.load(self.client.put(self.url, data={
+            "RequestInfo": {
+                "context": "_PARSE_.START.%s" % self.service_name,
+                "operation_level": {
+                    "level": "SERVICE",
+                    "cluster_name": self.cluster_name,
+                    "service_name": self.service_name,
+                }
+            },
+            "Body": {
+                "ServiceInfo": {"state": "STARTED"},
+            },
+        }))
+        return self
+
+    def stop(self, component_names=None):
+        """Stops this service."""
+        self.load(self.client.put(self.url, data={
+            "RequestInfo": {
+                "context": "_PARSE_.STOP.%s" % self.service_name,
+                "operation_level": {
+                    "level": "SERVICE",
+                    "cluster_name": self.cluster_name,
+                    "service_name": self.service_name,
+                }
+            },
+            "Body": {
+                "ServiceInfo": {"state": "INSTALLED"},
+            },
+        }))
+        return self
+
+
+class ConfigurationItemCollection(base.QueryableModelCollection):
+    @property
+    def url(self):
+        return self.parent.url
+
+    def create(self, tag=None, **kwargs):
+        """This collection requires you to PUT the entire collection every time you add to it"""
+        # support for create(properties='{"hadoop.proxyuser.xyz.groups" : "*"}')
+        # a PUT to "http://localhost:8080/api/v1/clusters/cluster"
+        # in format as: [{"Clusters":{"desired_config":[{"type":"core-site", "tag":"somrandchars",
+        #                                                "properties":{"hadoop.proxyuser.xyz.groups" : "*", ... }]}}]
+        new_tag = (tag if tag else str(time.time()))
+        new_item = {'Clusters': {'desired_config': [{'type': self.parent.type, 'tag': new_tag }]}}
+        previous_tag = self.parent.cluster.desired_configs[self.parent.type]['tag']
+        self.inflate()
+        previous_tag_item = None
+        for model in self._models:
+            if model.tag == previous_tag:
+                previous_tag_item = (model.items[0] if (model.items and len(model.items) > 0) else None)
+                break
+
+        def uniquify(previous_tag_item, key, value):
```

Comment:
Perhaps this just be moved into a function in `utils.py`? I'm also a bit unclear as to its purpose and how it handles attributes of type `dict`, `list`, and other. Can you add some documentation around the intended use? If it simplifies things, I'd prefer to just handle one kind of type (if possible) to just keep the complexity down.